src/actions/sources.js

```diff
@@ -40,37 +41,48 @@ import {
   removeSourcesFromTabList,
   removeSourceFromTabList,
   getTextSearchQuery,
-  getActiveSearch
+  getActiveSearch,
+  getGeneratedSource
 } from "../selectors";
 
 import type { Source, Location } from "../types";
 import type { ThunkArgs } from "./types";
-import type { State } from "../reducers/types";
 
 // If a request has been made to show this source, go ahead and
 // select it.
-async function checkSelectedSource(state: State, dispatch, source) {
-  const pendingLocation = getPendingSelectedLocation(state);
+function checkSelectedSource(source) {
+  return async ({ dispatch, getState }: ThunkArgs) => {
+    const pendingLocation = getPendingSelectedLocation(getState());
 
-  if (pendingLocation && !!source.url && pendingLocation.url === source.url) {
-    await dispatch(selectLocation({ ...pendingLocation, sourceId: source.id }));
-  }
+    if (pendingLocation && !!source.url && pendingLocation.url === source.url) {
+      await dispatch(
+        selectLocation({ ...pendingLocation, sourceId: source.id })
+      );
+    }
+  };
 }
 
-async function checkPendingBreakpoints(state, dispatch, sourceId) {
-  // source may have been modified by selectLocation
-  const source = getSource(state, sourceId).toJS();
-  const pendingBreakpoints = getPendingBreakpointsForSource(state, source.url);
-  if (!pendingBreakpoints.size) {
-    return;
-  }
-
-  // load the source text if there is a pending breakpoint for it
-  await dispatch(loadSourceText(source));
-  const pendingBreakpointsArray = pendingBreakpoints.valueSeq().toJS();
-  for (const pendingBreakpoint of pendingBreakpointsArray) {
-    await dispatch(syncBreakpoint(sourceId, pendingBreakpoint));
-  }
+function checkPendingBreakpoints(sourceId) {
+  return async ({ dispatch, getState }: ThunkArgs) => {
+    // source may have been modified by selectLocation
+    const source = getSource(getState(), sourceId).toJS();
+
+    const pendingBreakpoints = getPendingBreakpointsForSource(
+      getState(),
+      source.url
+    );
+
+    if (!pendingBreakpoints.size) {
+      return;
+    }
+
+    await dispatch(loadSourceText(source));
```

Comment:
Curious! Why does the source needs to be loaded for checking the pending breakpoints?

Reply:
glad you asked :) because pending breakpoints need the symbols to compute the ast locations. thus we need to load the file